src/displayUI.js

```diff
@@ -0,0 +1,164 @@
+  param_description.className = "display-data-p";
+  param_description.textContent = `Description: ${link.Description}`;
+
+  let param_link = document.createElement("a");
+  param_link.className = "display-data-p";
+  param_link.href = link.Link;
+  param_link.target = "_blank";
+  param_link.textContent = "Link to an API webpage";
+
+  individuals.append(
+    param_svg,
+    param_api,
+    param_category,
+    param_category,
+    param_description,
+    param_link
+  );
+  document.querySelector("#oneCategoryDisplayBox").append(individuals);
+}
+
+// Display of all Link elements
+export function displayCategoryLink(arrayOfLinks, category) {
+  arrayOfLinks.forEach(link => {
+    if (category === link.Category) {
+      displayOneLink(link);
+    }
+  })
+}
+
+// Change of color for Mark-As-Important feature
+function changeColor(el) {
+  el.classList.toggle('mark-important');
+}
+
+//Search through Categories and display them
+export function searchCategory() {
+  const searchBar = document.querySelector("#searchBar");
+
+  searchBar.addEventListener("input", (e) => {
+    e.preventDefault();
+    document.querySelector("#categories").innerHTML = "";
+    instanceOfMainClass.categories.forEach((el) => {
+      if (el.toLowerCase().includes(e.target.value.toLowerCase())) {
+        displayOneCategory(el, instanceOfMainClass);
+      }
+    });
+  });
+}
+
+// Search through Links and display them
+export function searchLinks() {
+  const searchBarLinks = document.querySelector("#searchBarLinks");
+
+  searchBarLinks.addEventListener("input", (e) => {
+    e.preventDefault();
+    
+    document.querySelector("#oneCategoryDisplayBox").innerHTML = "";
+
+    instanceOfMainClass.links.forEach((el) => {
+      if (el.API.toLowerCase().includes(e.target.value.toLowerCase())) {
+        displayOneLink(el);
+      }
+    });
+  });
+}
+
+//btnClear for Category search
+export function btnClear() {
+  let searchContent = document.querySelector("#searchBar");
+
+  if (searchContent.value !== "") {
+    searchContent.value = "";
+    document.querySelector("#categories").innerHTML = "";
+  } else {
+    searchContent.value = "";
+    document.querySelector("#categories").innerHTML = "";
+  }
+}
+
+//btnClear for Links search
+export function btnClearLinks() {
+  let searchContent = document.querySelector("#searchBarLinks");
+
+  if (searchContent.value !== "") {
+    searchContent.value = "";
+    document.querySelector("#oneCategoryDisplayBox").innerHTML = "";
+  } else {
+    searchContent.value = "";
+    document.querySelector("#oneCategoryDisplayBox").innerHTML = "";
+  }
+}
+
+// initialization of btnClear for Category
+document.querySelector("#btnClear").addEventListener("click", (e) => {
+  e.preventDefault();
+  btnClear(instanceOfMainClass);
+});
+
+// initialization of btnClear for Links
+document.querySelector("#btnClearLinks").addEventListener("click", (e) => {
+  e.preventDefault();
+  btnClearLinks(instanceOfMainClass);
```

Comment:
don't need to send instance since you imported instance and is available everywhere.